### PR TITLE
Remove v5 support

### PR DESCRIPTION
**Pull request by paultyng**
Should this be a major version bump of the provider? Maybe not.

The last release of v5 was over a year ago so I think this is probably safe to do?

### CODE REVIEW

**Comment by joshuaspence** [comment]
Just trying to understand the context of this change (I'm largely in favour regardless). Is this an effort to simplify the code / clean up or is there a bigger issue with v5? Unless there is a significant maintenance cost, I'd rather deprecate v5 support (in such a way that Terraform logs it as a warning) for a few months before removing support entirely. This is only because I have no idea what v5/V6 adoption numbers look like.

**Comment by paultyng** [comment]
Yeah mostly looking at just cleaning up the codebase. Can just leave this PR here and start with a warning PR instead, that's a good idea.

**Comment by paultyng** [comment]
Oh I forgot, I already have the deprecation messages:

https://github.com/paultyng/terraform-provider-unifi/blob/be8f77c17f0cd03db6eeac98a27fdd2d00b9df66/internal/provider/data_wlan_group.go#L15

https://github.com/paultyng/terraform-provider-unifi/blob/be8f77c17f0cd03db6eeac98a27fdd2d00b9df66/internal/provider/resource_wlan.go#L217

So they are already getting warnings.

I think the best thing is probably go to v1 and if someone really needs it we can always create a v0.x release in the future with 5.0 support or something if a hotfix were required.

**Comment by joshuaspence** [comment]
Those deprecations would only show if people are using v5-specific resources. I think someone managing `unifi_network` resources only, for example, wouldn't see any deprecation warnings.

**Comment by paultyng** [comment]
Yeah, I can't see any easy way to add those warnings without updating every read/create currently or switching from the lazy client. It may be useful to do so, I haven't checked in a while, but you used to need to defer connection in a provider to properly connect to things that may be configured as part of the run but thats pretty unlikely with the controller probably. For example someone who provisions a docker controller in the cloud and Terraform's against that in the same config. Maybe we should just unwind the lazy client so we can warn at configure time?

**Review by joshuaspence — DISMISSED** [review]
You can also get rid of the `unifi_wlan_group` in `examples/provider/test.tf`

**Comment by joshuaspence** [comment]
(I mostly just eyeballed this and did some grepping of my own)

File: docker-compose.yaml [hunk 1]
```diff
@@ -1,24 +1,24 @@
 ---
 services:
   unifi:
-    image: 'jacobalberty/unifi:${UNIFI_VERSION:?}'
```
Comment (joshuaspence) [inline docker-compose.yaml:4]: Ha, personally I prefer single quotes but I am okay either way as long as it's consistent. I have been meaning to setup `yamllint`

File: tools/tools.go [hunk 2]
```diff
@@ -1,3 +1,4 @@
+//go:build tools
```
Comment (joshuaspence) [inline tools/tools.go:1]: What does this do?

Comment (paultyng) [reply tools/tools.go:1]: Its for Go 1.17, its added automatically by `gofmt`

File: docker-compose.yaml (hunk 1 above)
Comment (paultyng) [reply docker-compose.yaml:4]: Yeah I just have `prettier` installed for work and got tired of fighting it 😭

**Pull request merged**